mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js

```diff
@@ -0,0 +1,144 @@
+import language from '../../language'
+
+const getValueToReturn = ({ nonOtherCheckboxItems, otherInputValue, isOtherChecked }) => {
+  const valueToReturn = { selectedValues: nonOtherCheckboxItems ?? [] }
+  if (isOtherChecked) {
+    valueToReturn.otherValue = otherInputValue
+  }
+
+  return valueToReturn
+}
+
+const CheckboxGroup = forwardRef(
+  ({ id, onBlur, onChange, options, shouldAddOtherOptionWithClarification, value }, ref) => {
+    const [nonOtherCheckboxItems, setNonOtherCheckboxItems] = useState([])
+    const [otherInputValue, setOtherInputValue] = useState(value?.otherValue ?? '')
+    const [isOtherChecked, setIsOtherChecked] = useState(!!value?.otherValue)
+
+    const _loadNonOtherCheckboxItems = useEffect(() => {
+      setNonOtherCheckboxItems(value?.selectedValues ?? [])
+    }, [value?.selectedValues])
+
+    const handleOtherCheckboxChange = () => {
+      const isOtherCheckedNewState = !isOtherChecked
+      setIsOtherChecked(isOtherCheckedNewState)
+      onChange(
+        getValueToReturn({
+          nonOtherCheckboxItems,
+          otherInputValue,
+          isOtherChecked: isOtherCheckedNewState
+        })
+      )
+    }
+    const handleOtherTextInputChange = (event) => {
+      const {
+        target: { value }
+      } = event
+      setOtherInputValue(value)
+      onChange(
+        getValueToReturn({
+          nonOtherCheckboxItems,
+          otherInputValue: value,
+          isOtherChecked
+        })
+      )
+    }
+    const handleNonOtherCheckboxChange = (itemValue) => {
+      const updateNonOtherCheckboxItems = [...nonOtherCheckboxItems]
+      const foundItemIndex = updateNonOtherCheckboxItems.indexOf(itemValue)
+
+      if (foundItemIndex > -1) {
+        updateNonOtherCheckboxItems.splice(foundItemIndex, 1)
+      } else {
+        updateNonOtherCheckboxItems.push(itemValue)
+      }
+
+      setNonOtherCheckboxItems(updateNonOtherCheckboxItems)
+      onChange(
+        getValueToReturn({
+          nonOtherCheckboxItems: updateNonOtherCheckboxItems,
+          otherInputValue,
+          isOtherChecked
+        })
+      )
+    }
+
+    const otherThanOtherCheckboxInputs = options.map((item) => (
+      <div key={item.value}>
+        <FormControlLabel
+          control={
+            <Checkbox
+              id={`${id}-${item.value}`}
+              value={item.value}
+              checked={nonOtherCheckboxItems.includes(item.value)}
+              onChange={() => {
+                handleNonOtherCheckboxChange(item.value)
+              }}
+              onBlur={onBlur}
+            />
+          }
+          label={item.label}
+        />
+      </div>
+    ))
+
+    return (
+      <FormGroup ref={ref}>
+        {otherThanOtherCheckboxInputs}
+        {shouldAddOtherOptionWithClarification ? (
+          <div>
+            <FormControlLabel
+              control={
+                <Checkbox
+                  id={`${id}-other`}
+                  value={'OTHER'}
```

Comment:
I was passing in 'Other' as the value instead of all caps - maybe a small thing but just want to make sure we are consistent
![Screen Shot 2022-05-18 at 2 13 35 PM](https://user-images.githubusercontent.com/26089140/169115897-5eb2b7d8-352a-4f7a-a877-bea1abaa282d.png)